pylint/checkers/classes.py

```diff
@@ -838,7 +842,7 @@ def leave_classdef(self, cnode):
         access to existent members
         """
         # check access to existent members on non metaclass classes
-        if self._ignore_mixin and cnode.name[-5:].lower() == "mixin":
+        if self._ignore_mixin and self._mixin_rgx.match(cnode.name.lower()):
```

Comment:
Using `mixin-class-rgx=.*[Mm]ixin` is confusing if the whole thing is case insensitive.  

(Although from the code you're using, it's not Case-insensitive, and changing the setting to `mixin-class-rgx=.*(MixinAPI|Base)` would never match your forced-lowercase classnames)

Reply:
Would love to set re.I on the Option. Any idea how I would go about that?

Reply:
I agree with @silasary. The fact that you use `.lower()` ruins the case sensibility of the chosen regexp.